internal/pkg/instrumentation/manager.go

```diff
@@ -398,11 +419,189 @@ func (m *Manager) loadProbes() error {
 	return nil
 }
 
+func (m *Manager) LoadProbe(i ProbeReference, name probe.ID, cfg Config) error {
+	m.logger.Info("loading probe", "name", name)
+
+	spec, err := i.probe.Spec()
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+
+	err = m.InjectProbeConsts(i.probe, spec)
+	if err != nil {
+		return errors.Join(err, m.cleanup())
+	}
+
+	c, err := utils.InitializeEBPFCollection(spec, m.collectionOpts)
```

Comment:
Should this function be moved to this package/file. This is the only place it is used.